src/components/DatePicker/DatePicker.tsx

```diff
@@ -210,12 +224,43 @@ export const DatePicker = ({
       setSelectedDates(
         new Date(currentDate.getFullYear(), currentDate.getMonth(), 1),
       )
+    } else if (mode === 'yearPicker') {
+      setSelectedDates(
+        new Date(currentDate.getFullYear(), 0, 1),
+      )
     }
   }
 
-  const isTodayOrAfter = Boolean(
-    selectedDates instanceof Date && selectedDates >= new Date(),
-  )
+  const isTodayOrAfter = useMemo(() => {
+    switch(mode) {
+      case 'dayPicker':
+        return Boolean(
+          selectedDates instanceof Date && (
+            endOfDay(selectedDates) >= endOfDay(new Date()) || endOfDay(selectedDates) >= maxDate
+          ),
+        )
+      case 'monthPicker':
+        return Boolean(
+          selectedDates instanceof Date && (
+            endOfMonth(selectedDates) >= endOfMonth(new Date())
+            || endOfMonth(selectedDates) >= maxDate
+          ),
+        )
+      case 'yearPicker':
+        return Boolean(
+          selectedDates instanceof Date && (
+            endOfYear(selectedDates) >= endOfYear(new Date()) || endOfYear(selectedDates) >= maxDate
+          ),
+        )
```

Comment:
We want to check if the selected date matches "Today/Current" term. In case of switching between dates, modes etc. in date picker, hooks and other states, we may end up where irrelevant parts of DateTimes do not match, ie. 01/15/24 00:00:00 vs 01/01/24 12:14:15 - when we compare months or years, both dates points to the same month/year. So there we compare endOf...